app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -44,8 +43,11 @@ def mint_doi(work)
         work.update(doi: [register_doi.identifier])
       end
 
-      def can_mint_for?(work)
-        work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited"
+      def cant_mint_for(work)
```

Comment:
Could you use the `?` suffix for methods like this, so that we know its a boolean result being returned. 

Why are you using a negative and not a positive that is inverted? I think using negative methods makes it harder keep track of whats going on.